src/js/controller.js

```diff
@@ -503,15 +503,14 @@ export const MediaUIStates = {
     textTracksEvents: ['addtrack', 'removetrack'],
   },
   MEDIA_SUBTITLES_SHOWING: {
-    get: function (controller) {
+    get: function (controller, event) {
       // NOTE: A bit hacky, but this ensures that HAS-style textTracks (e.g. from mux-video)
       // will also respect `defaultsubtitles` (CJP)
       if (
         controller.hasAttribute('defaultsubtitles') &&
-        !controller.hasAttribute(MediaUIAttributes.MEDIA_HAS_PLAYED) &&
```

Comment:
NOTE: changed the logic here. Since tracks can be added after media has started playback, particularly with autoplay + HAS (e.g. HLS), we need to approach this differently. Now, we'll end up re-applying default subtitles whenever a track is added or removed. This will result in slightly different behavior when, e.g., swapping out sources or more advanced cases of programmatically adding/removing text tracks.